surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/RunListenerAdapter.java

```diff
@@ -60,14 +60,14 @@
     public void testPlanExecutionStarted( TestPlan testPlan )
     {
         this.testPlan = testPlan;
+        failures.clear();
```

Comment:
ah, ok, this looks reasonable. It is called after the tests of the first phase, right?

Reply:
Correct. When one test plan finishes and the next starts the failures of the prior execution are cleared.